watcher/src/watcher.rs

```diff
@@ -118,11 +123,14 @@ impl Watcher {
     ///
     /// * `start` - starting block to sync.
     /// * `max_block_height` - the max block height to sync per archive url. If None, continue polling.
+    ///
+    /// Returns true if syncing has reached max_block_height, false if more blocks still need to be
+    /// synced.
     pub fn sync_signatures(
         &self,
         start: u64,
         max_block_height: Option<u64>,
-    ) -> Result<(), WatcherError> {
+    ) -> Result<bool, WatcherError> {
```

Comment:
This is part of a bugfix in `main.rs` that caused the loop to run indefinitely even when a max block was specified.